File: .github/security/semgrep-tests/security.js
function dynamicExecution(source) {
  // ruleid: gymapp-javascript-dynamic-code-execution
  return eval(source);
}

function cleartextUrl() {
  // ruleid: gymapp-hardcoded-cleartext-url
  return "http://example.invalid/api";
}

function secureUrl() {
  // ok: gymapp-hardcoded-cleartext-url
  return "https://example.invalid/api";
}

File: .github/security/semgrep-tests/security.kt
fun commandExecution() {
    // ruleid: gymapp-kotlin-process-execution
    Runtime.getRuntime().exec("id")
}

fun dangerousWebView(webView: android.webkit.WebView) {
    // ruleid: gymapp-kotlin-dangerous-webview-bridge
    webView.addJavascriptInterface(Any(), "bridge")
}

fun sensitiveLogging(token: String) {
    // ruleid: gymapp-kotlin-sensitive-logging
    android.util.Log.i("Auth", "accessToken=$token")
    // ok: gymapp-kotlin-sensitive-logging
    android.util.Log.i("Sync", "sync rejected")
}

fun cleartextUrl(): String {
    // ruleid: gymapp-hardcoded-cleartext-url
    return "http://example.invalid/api"
}

fun secureUrl(): String {
    // ok: gymapp-hardcoded-cleartext-url
    return "https://example.invalid/api"
}
